contracts/zero-ex/contracts/src/liquidity-providers/CurveLiquidityProvider.sol

```diff
@@ -0,0 +1,208 @@
+        outputToken.compatTransfer(recipient, boughtAmount);
+    }
+
+    /// @dev Trades token for ETH. The token must be sent to the contract prior
+    ///      to calling this function to trigger the trade.
+    /// @param inputToken The token being sold.
+    /// @param recipient The recipient of the bought tokens.
+    /// @param minBuyAmount The minimum acceptable amount of ETH to buy.
+    /// @param auxiliaryData Arbitrary auxiliary data supplied to the contract.
+    /// @return boughtAmount The amount of ETH bought.
+    function sellTokenForEth(
+        IERC20TokenV06 inputToken,
+        address payable recipient,
+        uint256 minBuyAmount,
+        bytes calldata auxiliaryData
+    )
+        external
+        override
+        returns (uint256 boughtAmount)
+    {
+        require(
+            !LibERC20Transformer.isTokenETH(inputToken),
+            "CurveLiquidityProvider/INVALID_ARGS"
+        );
+        boughtAmount = _executeSwap(
+            inputToken,
+            LibERC20Transformer.ETH_TOKEN,
+            minBuyAmount,
+            abi.decode(auxiliaryData, (CurveData))
+        );
+        // Every pool contract currently checks this but why not.
+        require(boughtAmount >= minBuyAmount, "CurveLiquidityProvider/UNDERBOUGHT");
+        recipient.transfer(boughtAmount);
+    }
+
+    /// @dev Quotes the amount of `outputToken` that would be obtained by
+    ///      selling `sellAmount` of `inputToken`.
+    function getSellQuote(
+        IERC20TokenV06 /* inputToken */,
+        IERC20TokenV06 /* outputToken */,
+        uint256 /* sellAmount */
+    )
+        external
+        view
+        override
+        returns (uint256)
+    {
+        revert("CurveLiquidityProvider/NOT_IMPLEMENTED");
+    }
+
+    /// @dev Perform the swap against the curve pool. Handles any combination of
+    ///      tokens
+    function _executeSwap(
+        IERC20TokenV06 inputToken,
+        IERC20TokenV06 outputToken,
+        uint256 minBuyAmount,
+        CurveData memory data
+    )
+        private
+        returns (uint256 boughtAmount)
+    {
+        uint256 sellAmount =
+            LibERC20Transformer.getTokenBalanceOf(inputToken, address(this));
+        if (!LibERC20Transformer.isTokenETH(inputToken)) {
+            inputToken.approveIfBelow(data.curveAddress, sellAmount);
+        }
+
+        (bool success, bytes memory resultData) =
+            data.curveAddress.call
+                { value: LibERC20Transformer.isTokenETH(inputToken) ? sellAmount : 0 }
+                (abi.encodeWithSelector(
+                    data.exchangeFunctionSelector,
+                    data.fromCoinIdx,
+                    data.toCoinIdx,
+                    // dx
+                    sellAmount,
+                    // min dy
+                    minBuyAmount
+                ));
+        if (!success) {
+            resultData.rrevert();
+        }
+        if (resultData.length == 32) {
```

Comment:
Given that any Curve function can be called, I wonder if this could be tricked by calling `get_dy` instead of `exchange`. IIRC the trailing `minBuyAmount` parameter would just ignored if it's passed into `get_dy`.

```
// exchange tokens
function exchange(
    uint128 fromCoinIdx,
    uint128 toCoindIdx,
    uint256 sellAmount,
    uint256 minBuyAmount
) returns (uint256 buyAmount);

// simulate exchanging tokens
function get_dy(
    uint128 fromCoinIdx,
    uint128 toCoindIdx,
    uint256 sellAmount
) returns (uint256 buyAmount);
```

Reply:
Sure, but why would you. You could also just pass in any evil contract address in `data.curveAddress` as well. The balance check in LPFeature is really the only line of defense for the taker.